### DISCUSSION TITLE
[New Feature] Test old Members

### DISCUSSION THREAD

**Discussion by Mccpie01** [J-Rios/TLG_JoinCaptchaBot#208] [Ideas]
I like to add a mode that would test old members of a group to see if they are alive or zombies
If no interaction with group for 7 days , send a captcha test and remind them for 7 day to solve it. 
If they dont, kick them out

**Comment by J-Rios** [comment]
Hello, I really like the idea.

Some points to mention and discuss:

- Should be a configurable mode that an Admin of a group can enable or disable (disabled by default).

- Regarding your proposal point, "remind them for 7 day to solve it"... The easy implementation for that requires to keep in RAM the users that are pending to solve a captcha, and also CPU load for checking if timeout for them has come. That's inefficient usage of machine resources when the Bot is handling thousands/millions of users at the same time. So we should change the approach or find a better solution/procedure that is not so CPU/RAM intensive (for example, storing old users pending to solve a captcha in hard-disk files, and just read and check them one time per day).

- Commonly, we can say that only ~20% (or even less) members of a group are active members, so ~80% are not active... In a group with 1000 members, it means that 800 members are not-active, and after 7 days, all of them needs to be managed to request them to solve a captcha, and we must not request to solve the captcha to all of them at the same time (it will cause a lot of noise in group chat).

For previous problematics, I suggest to change the approach, perhaps is better to just send some general info message in the group one time per day advertising members that any member that wont publish a message in 7 days will be removed from the group...

New proposal:

- New mode to automatically remove not-active members of a group after 7 days.
- The mode should be configurable (disabled by default).
- The Bot will send a message one time per day advertising that inactive members will be removed after 7 days.

**Reply by Mccpie01** [reply]
Good points. I like it
